lib/plan_executor/app.rb

```diff
@@ -78,15 +69,24 @@ def validate_schema(schema, body)
       error = validate_schema(@schema, body)
       return [400, error.to_json] unless error.nil?
 
+      # Create a basic executor, leave concurrency up to Orchestrator.
+      # This is just a class variable for testing
+      executor = PlanExecutor::Executor.new(0, body['job_id'])
+      # Use an empty inventory until we figure out where this data comes from.
+      inventory = Bolt::Inventory.new(nil)
+      pal = Bolt::PAL.new(@modulepath, nil)
```

Comment:
Why did pal move here? Is it not threadsafe? I don't see any obvious mutable state in the PAL class.

Reply:
It makes this comment a lie. https://github.com/puppetlabs/bolt/blob/master/lib/bolt/pal.rb#L40

There are far bigger problems with how we're using PAL at this point though since the setting method is not threadsafe regardless of how many instances there are.